### PR TITLE
add create/get aws instance success

### PR DESCRIPTION
**Pull request by fanux**
Signed-off-by: fanux <fanux@sealos.io>

### CODE REVIEW

File: controllers/infra/drivers/reconcile.go [hunk 1]
```diff
@@ -28,7 +28,8 @@ func (a *Applier) ReconcileInstance(infra *v1.Infra, driver Driver) (*v2.Cluster
 	// get current hosts
 	tag := infra.GetInstancesTag()
 	var err error
-	a.currentHosts, err = driver.GetInstancesByLabel(common.InfraInstancesLabel, tag)
+	hosts, err := driver.GetInstancesByLabel(common.InfraInstancesLabel, tag, infra)
+	a.currentHosts = append(a.currentHosts, *hosts)
 	if err != nil {
 		return nil, fmt.Errorf("failed to query instances: %v", err)
 	}
```
Comment (zzjin) [inline controllers/infra/drivers/reconcile.go:35]:
```suggestion
	hosts, err := driver.GetInstancesByLabel(common.InfraInstancesLabel, tag, infra)
	if err != nil {
		return nil, fmt.Errorf("failed to query instances: %v", err)
	}
	a.currentHosts = append(a.currentHosts, *hosts)
```

**Comment by cuisongliu** [comment]
Https://kubebuilder.io/reference/using-finalizers.html Do you want to optimize this? I wrote a sdk before that can be used directly.  https://github.com/labring/endpoints-operator/blob/main/library/controller/controller.go

File: controllers/infra/drivers/reconcile.go (hunk 1 above)
Comment (fanux) [reply controllers/infra/drivers/reconcile.go]: done

**Comment by fanux** [comment]
> [Https://kubebuilder.io/reference/using-finalizers.html](Https://kubebuilder.io/reference/using-finalizers.html) Do you want to optimize this? I wrote a sdk before that can be used directly. https://github.com/labring/endpoints-operator/blob/main/library/controller/controller.go

yes I will

**Review by cuisongliu — APPROVED** [review]

**Pull request merged**